tests/test_unit/approval_client/client.py

```diff
@@ -124,7 +127,10 @@ def external_id(
         try:
             return self._reverse_cache[id] if isinstance(id, int) else [self._reverse_cache[i] for i in id]
         except KeyError:
-            raise RuntimeError(f"{type(self).__name__} does not support reverse lookup before lookup")
+            if self._allow_reverse_lookup:
+                return "".join(random.choices(string.ascii_letters + string.digits, k=10))
```

Comment:
The name `_allow_reverse_lookup` was confusing, I read it as `__generate_random_external_id_on_lookup_failure`

Reply:
Thanks. I think there is some renaming needed in the client in the code base as well, but will save that for another PR. Did you suggestion for now.